File: backend/app/schemas/admin/system.py
from typing import Any, List, Optional
from datetime import datetime

from pydantic import BaseModel, Field, ConfigDict


class SystemSettingItem(BaseModel):
    key: str
    label: str
    value: Any
    value_preview: Optional[str] = None
    type: str = "text"
    editable: bool = False
    pending: bool = False
    help_text: Optional[str] = None


class SystemSettingsCategory(BaseModel):
    id: str
    title: str
    description: str
    settings: List[SystemSettingItem] = Field(default_factory=list)


class SystemSettingsResponse(BaseModel):
    generated_at: datetime
    categories: List[SystemSettingsCategory]

File: frontend/src/types/admin/system.ts
export interface SystemSettingItem {
  key: string;
  label: string;
  value: unknown;
  value_preview?: string | null;
  type: string;
  editable: boolean;
  pending: boolean;
  help_text?: string | null;
}

export interface SystemSettingsCategory {
  id: string;
  title: string;
  description: string;
  settings: SystemSettingItem[];
}

export interface SystemSettingsResponse {
  generated_at: string;
  categories: SystemSettingsCategory[];
}
